File: 0136-single-number/0136-single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) 
    {
        int size = nums.size();
        map <int,int> freq;
        
        int result = 0;
        
        for (int i = 0; i < size; i++)
            if (freq.find(nums[i]) == freq.end())
                freq[nums[i]] = 1;
            else
                freq[nums[i]] += 1;
        
        for (int i = 0; i < size; i++)
            if (freq[nums[i]] == 1)
                return nums[i];
          
        return 0;
    }
};

File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # To find the single element in an array where every element 
        # appears twice except for one, you can use the XOR (exclusive OR) 
        # operation. XORing a number with itself results in zero, so when you XOR 
        # all the numbers in the array together, the duplicate elements cancel each 
        # other out, leaving only the single element.
        
        res = 0
        
        for num in nums:
            res ^= num
        return res